SOFTWARE/AfterlifeLightKit/Lights.h

```diff
@@ -0,0 +1,101 @@
+#define FASTLED_INTERNAL //remove annoying pragma messages
+#include "FastLED.h"
+
+#include "ConfigManager.h"
+
+#ifndef Lights_h
+#define Lights_h
+
+class Lights {
+    public:
+        // Constructor
+        Lights();
+
+        // Public Methods
+        void init(ConfigManager configManager);
+        void setMode(MODES mode);
+        void update();
+
+        void tickPowercell();
+        void tickCyclotron();
+    private:
+        // Private Methods
+        bool _checkTimer(unsigned long &previous, unsigned long interval);
+        int _getNextPixel(int currentPixel, int stripLength, Directions direction);
+
+        // Private Properties
+        int _powercellLength;
+        int _cyclotronLength;
+        unsigned long _previousPowercellMillis = 0;
+        unsigned long _previousCyclotronMillis = 0;
+        int _powercellInterval;
+        int _cyclotronInterval;
+        int _currentPowercell = 0;
+        int _currentCyclotron = 0;
+
+        bool _isDirty = false;
+
+        ConfigManager _configManager;
+        Configuration _configuration;
+        HardwareConfiguration _hardwareConfiguration;
+        Settings _settings;
+        MODES _mode;
+        /**
+         * Have to define these with 100 as we can't seem to dynamically set the strip length
+         * So 20 should be more than enough for the Powercell, and 100 for the Cyclotron
+         */
+        CRGB _powercellLEDS[20];
+        CRGB _cyclotronLEDS[100];
```

Comment:
I didn't want to hard-code these LED strip lengths, but it seems to be required for memory allocation and can't be initialized dynamically from configuration, so I set them higher than what most people would need. I'm now thinking we just go back to `16` and `60` respectively, define them in `Settings.h`, and let people recompile if they need to. You can always change the software length from configuration, just not the hardware length.

Reply:
I certainly think it is fine to choose a safe margin value for each. Should anyone need something way more, they should edit the software because they are gonna have something weird going on anyways.